src/controllers/ExperimentViewController.ts

```diff
@@ -113,6 +131,43 @@ export class ExperimentViewController {
     }
   }
 
+  /**
+   * React to any state changes
+   */
+  private eventListener(): () => void {
+    return store.subscribe(() => {
+      // Get the latest state
+      const state = store.getState()
+
+      // Get the latest entry from our event queue
+      const latestEvent = latestEventSelector(state)
+
+      // TODO: Change this back to 1 hour
+      const SUSPENDED_TIMEOUT = 60
+      const STRICT_SUSPENED_TIMEOUT = 30
+
+      // Check if user has suspened app in crucial stage
+      if (latestEvent?.eventType == 'SuspensionPeriod') {
+        // If app was suspened longer than timeout threshold...
+        const suspenedTime = Number(latestEvent.value)
+        if (
+          suspenedTime > SUSPENDED_TIMEOUT ||
+          (this.focusModeEnabled && suspenedTime > STRICT_SUSPENED_TIMEOUT)
+        ) {
+          this.screenOutParticipant()
+        }
+      }
+    })
+  }
+
+  /**
+   * Focus mode allows the experiment to determine whether a user suspending the app is detrimental
+   * to the data intergrity.
+   */
+  setFocusMode(value: boolean) {
+    this.focusModeEnabled = value
```

Comment:
I'm worried that we've now introduced mutable state to our React app... I feel like we should consider revising the structure of the experiment controller. It's starting to feel really foreign in a React app.

Maybe we could do something more akin to how React Router handles displaying different screens. We can talk about this on Slack/elsewhere.

Reply:
This has been discussed via Slack and we will come back to restructuring this.